src/runtime/virtcontainers/utils/utils.go

```diff
@@ -390,3 +391,61 @@ outer:
 
 	return nil
 }
+
+func MkdirAllWithInheritedOwner(path string, perm os.FileMode) error {
+	var sb strings.Builder
+	sb.WriteString(path)
+	// Append '/' to the end if the path doesn't end with '/'
+	if !strings.HasSuffix(path, string(os.PathSeparator)) {
+		sb.WriteRune(os.PathSeparator)
```

Comment:
I'm a bit confused by this function tbh. Why can't you use `[file]path.Join()` ooi?

Reply:
The purpose is to format the path to `/foo/bar/` so that the following code doesn't need to handle the case where a path doesn't end with `/`.

Reply:
But `filepath.Join()` handles that scenario automatically doesn't it?